### PR TITLE
Feature/edit tags tags types

### PR DESCRIPTION
**Pull request by michal1990**

### CODE REVIEW

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivity.java [hunk 1]
```diff
@@ -92,12 +92,19 @@ public void prepareView() {
...
                 activityEditBinding.setViewModel(editViewModel);
             } catch (IOException e) {
                 Log.d(TAG, String.valueOf(e.getMessage()));
                 //TODO log exception to crashlitycs on else.
             }
         } //TODO log exception to crashlitycs on else.
     }
+
+    @Override
+    public void openEditDialog(Tag tag) {
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivity.java]: Its better to encapsulate interfaces as a private fields. 
Class logic is safe and not exposed for changes from outside.

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditDialog.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        return (EditDialog) Fragment.instantiate(context, FRAGMENT_TAG, bundle);
+    }
+
+    private Tag tag;
+
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        if (!extractTag()) {
+            throw new IllegalStateException("Tag is null");
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditDialog.java]: This should be handled

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditViewModel.java [hunk 3]
```diff
@@ -12,15 +12,18 @@
...
     private final EditCompoundRecyclerView editCompoundRecyclerView;
     private final TagsManager tagsManager;
+    private final EditTagListener editTagListener;
 
     EditViewModel(String title,
                   String imageUrl,
                   EditCompoundRecyclerView editCompoundRecyclerView,
-                  TagsManager tagsManager) {
+                  TagsManager tagsManager,
+                  EditTagListener editTagListener) {
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditViewModel.java]: editTagListener is redundant, listener is enough,
tagsManager -> manager
EditCompoundRecyclerView -> recyclerView (not part of this commit)

**Review by BartSoftware — CHANGES_REQUESTED** [review]

**Review by kejmil01 — APPROVED** [review]

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditDialog.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        if (extractTag()) {
+            prepareView();
+        } //TODO log exception to crashlitycs on else.
+
+        return super.onCreateDialog(savedInstanceState);
+    }
+
+    private boolean extractTag() {
+        if (getArguments() != null && !getArguments().isEmpty()) {
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditDialog.java:39]: When there are more then one condition I prefer to check them with variables like: 
Object object = getArguments();
 if ( object!= null && !object.isEmpty()) {
             Parcelable extra = object.getParcelable(EXTRA_TAG);

But its not a blocker (just my fantasy ;))

**Review by BartSoftware — APPROVED** [review]
